quesma/table_resolver/model.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package table_resolver
+
+import (
+	"fmt"
+	"quesma/logger"
+	"strings"
+)
+
+type Decision struct {
+	// obvious fields
+	IsClosed bool
+	Err      error
+	IsEmpty  bool
+
+	// which connector to use, and how
+	UseConnectors []ConnectorDecision
+
+	// who made the decision and why
+	Message      string
+	ResolverName string
+}
+
+func (d *Decision) String() string {
+
+	var lines []string
+
+	if d.IsClosed {
+		lines = append(lines, "Returns a closed index message.")
+	}
+
+	if d.IsEmpty {
+		lines = append(lines, "Returns an empty result.")
+	}
+
+	if d.Err != nil {
+		lines = append(lines, fmt.Sprintf("Returns error: '%v'.", d.Err))
+	}
+
+	for _, connector := range d.UseConnectors {
+		lines = append(lines, connector.Message())
+	}
+
+	lines = append(lines, fmt.Sprintf("%s (%s).", d.Message, d.ResolverName))
+
+	return strings.Join(lines, " ")
+}
+
+type ConnectorDecision interface {
+	Message() string
+}
+
+type ConnectorDecisionElastic struct {
+	// TODO  instance of elastic connector
+}
+
+func (*ConnectorDecisionElastic) Message() string {
+	return "Pass to Elasticsearch."
+}
+
+type ConnectorDecisionClickhouse struct {
+	// TODO  instance of clickhouse connector
+
+	ClickhouseTableName string
+	Indexes             []string
```

Comment:
Are these indexes elastic ones?

Reply:
These are matched indexes. I'll change the name.

Reply:
So maybe cleaner would be to change name to `ClickhouseIndexes` or `ClickhouseTables`?